src/otx/core/model/module/segmentation.py

```diff
@@ -31,7 +31,7 @@ def __init__(
         super().__init__(otx_model, optimizer, scheduler, torch_compile)
         metric_params = {
             "task": "multiclass",
-            "num_classes": otx_model.model.decode_head.num_classes,
+            "num_classes": otx_model.config.get("decode_head", {}).get("num_classes", None),
```

Comment:
Is this can be nullable?

https://github.com/Lightning-AI/torchmetrics/blob/3514d715ef42b8dc0c8d96cb3d581caba512fbd7/src/torchmetrics/classification/jaccard.py#L462

It seems that we cannot pass `None` for `multiclass`. Please try to fail here.

Reply:
We can raise an exception in case of "num_classes" is None

Reply:
Please raise an exception in that case with a comprehensible message such as `msg = 'if num_classes=None, cannot get JaccardIndex("multiclass") metric'`.

Reply:
[done](https://github.com/openvinotoolkit/training_extensions/pull/2748/commits/fa36076ef72b7ad990a68b70db33ed4eb007a889)